parse DataMessages as byte[] to fix Unicode issues

eighthave opened the pull request:
The original code was first parsing the message body into a `String`, then basing the TLV parsing off `String.length()`.  That works for ASCII, where one char is one byte, but totally fails for non-ASCII, since basically all non-latin chars are more than one byte.

This also has the advantage of removing the trailing null char from the end of the message body itself.

This should fix #46 for @aelmahmoudy

aelmahmoudy commented:
Thanks ! Just tested Xabber with this patch for otr4j and it worked !

cobratbq commented on src/main/java/net/java/otr4j/session/Session.java:
```diff
@@ -619,13 +609,28 @@ private String handleDataMessage(DataMessage data) throws OtrException {
+                // get the message body without the trailing 0x00, expect UTF-8 bytes
+                String decryptedMsgContent;
+                try {
+                    decryptedMsgContent = new String(dmc, 0, tlvIndex, "UTF-8");
```
If you use SerializationUtils.UTF8 here, you do not need to catch UnsupportedEncodingException. Makes the whole thing more readable.

eighthave replied:
Unfortunately, there is no equivalent `String` constructor that accepts a `CharSet`, which is what `SerializationUtils.UTF8` is.

eighthave replied:
I changed it to use `SerializationUtils.UTF8.name()` which is as good as its going to get, I think.  But the `try` block will have to stay.

eighthave replied:
Actually, I was totally wrong, switching to `SerializationUtils.UTF8`.  I looked for the right `String` constructor originally but for some reason, I concluded it wasn't there.

cobratbq commented:
Could you add a unit test for this. This is a great regression test for something that's hard to find otherwise.

Maybe we can extract the code for determining the start position of TVLs and move that to SerializationUtils?

eighthave committed:
parse DataMessages as byte[] to fix Unicode issues

The original code was first parsing the message body into a String, then
basing the TLV parsing off of the length of that String.  That works for
ASCII, where one char is one byte, but totally fails for non-ASCII, since
basically all non-latin chars are more than one byte.

This also has the advantage of removing the trailing null char from the end
of the message body itself.

closes https://github.com/otr4j/otr4j/issues/46
closes https://github.com/otr4j/otr4j/pull/50

eighthave commented:
As far as I can tell, this is the only spot in the code that needs to split off the TLVs from the rest of the message, so no need to move that stuff to a method.

As for unit tests, I have unicode related tests in another request: #48
